src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    public static String getCommand(String line) {
+        return line.split(" ")[0];
+    }
+
+    public static String getTaskDescription(String command, String line) throws DukeException {
+        if (line.split(command).length < 2 || line.split(command)[1].equals(" ")) {
+            throw new DukeException("☹ OOPS!!! Missing duke.task description!");
+        } else {
+            switch (command) {
+            case COMMAND_TODO:
+                return line.substring(line.indexOf(' ') + 1);
+            case COMMAND_DEADLINE:
+            case COMMAND_EVENT:
+                if (!line.contains("/by ") && command.equals(COMMAND_DEADLINE)) {
+                    throw new DukeException("☹ OOPS!!! Missing or incorrect /by statement");
+                }
+                if (!line.contains("/at ") && command.equals(COMMAND_EVENT)) {
+                    throw new DukeException("☹ OOPS!!! Missing or incorrect /at statement");
+                }
+                return line.substring(line.indexOf(' ') + 1, line.indexOf('/') - 1);
+            default:
+                return null;
+            }
+        }
+    }
+
+    public static String getTaskDate(String command, String line) {
+        switch (command) {
+        case COMMAND_DEADLINE:
+            return line.split("/by ")[1];
+        case COMMAND_EVENT:
+            return line.split("/at ")[1];
+        default:
+            return null;
+        }
+    }
+
+    public static int getDoneIndex(String command, String line) throws DukeException {
+        if (line.split(command).length < 2 || line.split(command)[1].equals(" ")) {
+            throw new DukeException("☹ OOPS!!! Missing index of duke.task!");
+        }
+        try {
+            return Integer.parseInt(line.split(" ")[1]) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("☹ OOPS!!! No integer index detected!");
+        }
+    }
+
+    public static void main(String[] args) {
+        printGreetingMessage();
+
+        // Initialize new instance of a TaskList object
+        TaskList list = new TaskList();
+
+        // Initialize Scanner and read in user input
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+        String command = getCommand(line);
+
+        while (!command.equals(COMMAND_BYE)) {
+            printHorizontalLine();
+            try {
+                switch (command) {
+                case COMMAND_LIST:
+                    list.printList();
+                    break;
+                case COMMAND_DONE:
+                    if (getDoneIndex(command, line) + 1 > list.getListLength()) {
+                        throw new DukeException("☹ OOPS!!! Index specified is out of list size!");
+                    }
+                    list.markAsDone(getDoneIndex(command, line));
+                    break;
+                case COMMAND_TODO:
+                    list.addToList(new ToDo(getTaskDescription(command, line)));
```

Comment:
Maybe can separate the printing done in `addToList` method with the action of adding to the TaskList? Like one call to add task to TaskList data structure, and another to print that the task has been added to list?